src/core/namespace.c

```diff
@@ -287,10 +288,15 @@ static int apply_mount(
 
                 /* First, get rid of everything that is below if there
                  * is anything... Then, overmount it with an
-                 * inaccessible directory. */
+                 * inaccessible path. */
                 umount_recursive(m->path, 0);
 
-                what = "/run/systemd/inaccessible";
+                r = lstat(m->path, &inaccessible_target);
```

Comment:
We can do use `stat()`, and document that symlinks are resolved from the host's root directory. Then it is the user that need to take care of the correctness of those link. Also We should return the error from `stat()` to handle loops, to many links, or other issue.